backend/cognito/cognito.template.yml

```
@@ -32,8 +38,37 @@ Globals:
     Timeout: 10
     Runtime: nodejs18.x
     PermissionsBoundary: !If [ UsePermissionsBoundary, !Ref PermissionsBoundary, !Ref AWS::NoValue ]
+    Environment:
+      Variables:
+        NOTIFY_API_KEY: !Ref NotifyApiKey
+        SECURITY_CODE_TEXT_MESSAGE_TEMPLATE: !Ref SecurityCodeTextMessageTemplate
 
 Resources:
+  CustomSenderEncryptionKey:
+    # checkov:skip=CKV_AWS_7: "Ensure rotation for customer created CMKs is enabled"
+    Type: AWS::KMS::Key
+    Properties:
+      KeyPolicy:
+        Version: "2012-10-17"
+        Id: "default_policy"
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            Action: 'kms:*'
+            Resource: '*'
+          - Effect: Allow
+            Principal:
+              Service: cognito-idp.amazonaws.com
+            Action: kms:CreateGrant
+            Resource: arn:aws:kms:eu-west-2:494650018671:key/375918d9-c680-44ff-8db2-14c1df0f10b8
+            Condition:
+              StringEquals:
+                aws:SourceAccount: "494650018671"
+              ArnLike:
+                aws:SourceArn: arn:aws:cognito-idp:eu-west-2:494650018671:userpool/eu-west-2_1shMc94As
+
```

```suggestion
              StringEquals:
                aws:SourceAccount: !Ref AWS::AccountId
              ArnLike:
                aws:SourceArn: arn:aws:cognito-idp:...:...:userpool/*
```
Use pseudo params and give access to all user pools in the account since we can't have a circular dependency.

---

addressed